.github/workflows/ci-rs.yml

```diff
@@ -53,6 +77,8 @@ jobs:
         run: cargo bench --verbose --no-run --workspace --all-features
 
   tests:
+    needs: changes
+    if: ${{ needs.changes.outputs.rust == 'true' }}
```

Comment:
It may be an issue that this is a "required" check for PRs, when it is sometimes skipped. Not sure what the best fix for this is; TKET does a [hack](https://github.com/CQCL/tket/blob/5ed1c366b419e0de03c581482ef1e94d12653a12/.github/workflows/build_and_test.yml#L580) where it creates a new "required" check that is a meta-check that everything succeeded or was skipped.

Reply:
I added the meta-check. I'll need to edit the required check name in the repo config before merging this PR.